### DISCUSSION TITLE
Splash screen on Android

### DISCUSSION THREAD

**Discussion by zach-xhx** [airsdk/Adobe-Runtime-Support#3497] [Ideas]
Would we ever be able to set a custom startup splash screen on Android instead of the solid black screen that can last for over 10 seconds on older phones? I know it used to be recommended to change app_entry/res/values/styles.xml, but that does not seem to work anymore. It can reliably change the background to another solid color, but it does not seem to let you set an image at all.

Could it hypothetically be possible for someone to make an ANE that allows custom splash screens on Android?

**Comment by marchbold** [comment]
Have you looked into using this approach: 

https://github.com/airsdk/Adobe-Runtime-Support/issues/1503#issuecomment-1016008239

**Comment by AndersonAge** [comment]
I tried to perform the above procedures for a background:

splash_background.xml in drawable:
```xml
    <item>
        <bitmap
            android:gravity="fill"
            android:src="@drawable/splash_image"/>
    </item>
```
styles.xml in values:
```xml
<resources>
    <style name="Theme.NoShadow" parent="android:style/Theme.NoTitleBar.Fullscreen">
        <item name="android:windowBackground">@drawable/splash_background</item>
</resources>
```
This only works on older devices, especially with the older Android version.
```xml
<uses-sdk android:minSdkVersion="21" android:targetSdkVersion="35" />
```
I also added the following folders: values, values-v31, values-v32, values-v32, values-v33, values-v34, values-v35 
and also: 
drawable, drawable-hdpi, drawable-mdpi, drawable-nodpi, drawable -xhdpi, drawable-xxhdpi, drawable-xxxhdpi

I'm wrong about what? Gaalxy S24 Ultra It never shows a slapscreen, it shows my app icon and starts it.
On the Galaxy S7 Edge, the splash screen appears.

Thank You!

**Reply by marchbold** [reply]
You need a `Theme.App.Splash` in your styles.xml. Have a read of the comment I linked above.

**Reply by AndersonAge** [reply]
> You need a `Theme.App.Splash` in your styles.xml. Have a read of the comment I linked above.


```html
<uses-sdk android:minSdkVersion="21" android:targetSdkVersion="35" />
``` 

My Styles.xml: (folder: /res/values/ and /res/values-v35

```html
<?xml version="1.0" encoding="utf-8"?>
<resources>
    <style name="Theme.NoShadow" parent="Theme.App.Splash"></style>
    <style name="Theme.App.Splash" parent="android:style/Theme.NoTitleBar.Fullscreen">
        <item name="android:windowBackground">@drawable/splash_background</item>
    </style>
</resources>
``` 


my splash_background.xml: (folder: /res/drawable)

```html
<?xml version="1.0" encoding="utf-8"?>
<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
    <item>
        <bitmap android:gravity="fill" android:src="@drawable/splash_image"/>
    </item>
</layer-list>
``` 

In drawable folder have splash_background.xml and splash_image.png

It works on older devices (Galaxy S7 Edge), it works fine on LDPlayer, but it doesn't show on Galaxy S24 ultra, Galaxy S21 ultra and Galaxy S10+. 
When you open the app, it only shows the app icon.
What do I need to do to get the galaxy s24 ultra to get the splash screen?


Galaxy S7 Edge: Android 8.0
Galaxy S10+: Android 12.0
S21 Ultra and S24 Ultra: Android 14.0

**Reply by marchbold** [reply]
Why are you using `values-v35`? This will only apply to devices running API 35+. What version is on the devices you are having issues with?

**Reply by AndersonAge** [reply]
> Why are you using 'values-v35'? This will only apply to running API 35+ devices. What version is on the devices you are having issues with?

android:targetSdkVersion="35"

Google requires version 34 or higher.

I tried with V31 but it doesn't work.

**Reply by marchbold** [reply]
Your target sdk is fine, but the `values-vXX` folder only applies those styles to vXX or higher. So you can keep your target sdk as is, but use `values-v31`

**Reply by AndersonAge** [reply]
> Seu sdk de destino está bom, mas a `values-vXX`pasta aplica apenas esses estilos ao vXX ou superior. Então você pode manter seu sdk de destino como está, mas use`values-v31`

Styles.xml: (folder: /res/values/ and /res/values-v31 now... (I even tried other values, none of them worked)
splash screen not work... show icons only

I think it has some permission, because on Android 12+ it doesn't work.

IN Galaxy S7 with android 8.0 and LDPLayer work!

Galaxy s24 ultra, S21 Ultra and S10+ dont work.

**Reply by AndersonAge** [reply]
> Seu sdk de destino está bom, mas a `values-vXX`pasta aplica apenas esses estilos ao vXX ou superior. Então você pode manter seu sdk de destino como está, mas use`values-v31`

If you want, I can test your application and see if the Splash Screen starts.
Because on mine, the background splash screen doesn't work. It could be that Android no longer accepts backgrounds.
